src/main/scala/it/pagopa/interop/apigateway/api/impl/GatewayApiServiceImpl.scala

```diff
@@ -127,11 +154,11 @@ final case class GatewayApiServiceImpl(
         case _                                            => Future.failed(Forbidden)
       }
       (prod, cons) = params
-      agreementState <- state.traverse(AgreementManagementApiAgreementState.fromValue(_).toFuture)
-      rawAgreements  <- agreementManagementService.getAgreements(prod, cons, eserviceId, descriptorId, agreementState)(
+      agreementStates <- parseArrayParameters(states).traverse(AgreementManagementApiAgreementState.fromValue).toFuture
+      rawAgreements <- agreementManagementService.getAgreements(prod, cons, eserviceId, descriptorId, agreementStates)(
         contexts
       )
-      agreements = rawAgreements.map(_.toModel)
+      agreements    <- rawAgreements.traverse(_.toModel).toFuture // * DRAFT was removed from open-api so it can't fail
```

Comment:
@galales this comment or custom error handling 😇

Reply:
I'm fine also without the comment 😁 
Just a note, this actually may fail if the user does not filter by `states` and the returned list contains a draft.
To avoid this we should default to a fixed agreement states list if the given is empty

Reply:
How about now?